analyzers/tests/SonarAnalyzer.Test/TestCases/UseShortCircuitingOperator.cs

```diff
@@ -15,5 +18,21 @@ public UseShortCircuitingOperator()
 
             var i = 1 | 2;
         }
+
+        public void ExtendedText(bool parameter)
+        {
+            var local = true;
+            var a = true | !true;                  // Noncompliant {{Correct this '|' to '||'.}}                                                                              Right hand side detected as constant
+            a = true | parameter;                  // Noncompliant {{Correct this '|' to '||'.}}                                                                              Right hand side detected as parameter reference
+            a = true | local;                      // Noncompliant {{Correct this '|' to '||'.}}                                                                              Right hand side detected as local reference
+            a = true | field;                      // Noncompliant {{Correct this '|' to '||'.}}                                                                              Right hand side detected as field reference
+            a = true | Property;                   // Noncompliant {{Correct this '|' to '||'.}}                                                                              Right hand side detected as property reference
+            a = true | ReturnSomeBool();           // Noncompliant {{Correct this '|' to '||' and extract the right operand to a variable if it should always be evaluated.}} Right hand side detected as possible side effect
+            a = true | !parameter;                 // Noncompliant {{Correct this '|' to '||' and extract the right operand to a variable if it should always be evaluated.}} Right hand side detected as possible side effect
+            a = true | (parameter ? true : false); // Noncompliant {{Correct this '|' to '||' and extract the right operand to a variable if it should always be evaluated.}} Right hand side detected as possible side effect
+        }
+
+        private bool ReturnSomeBool() =>
+            DateTime.Now.Second % 2 == 0;
```

Comment:
If we don't care about the result of this private method, why not always return `true` or `false`? Having `DateTime.Now` in a test makes the test non-deterministic. While the result of the test may not become itself flaky, I would still avoid any source of non-determinism.

Also, `DateTime.Now` can raise an exception in some "hostile contexts": e.g. `TimeZoneNotFoundException`.

Reply:
I want to make sure, that interprocedural code detection is not able to determine the result at compile time. I used `Environment.Is64BitOperatingSystem` instead now.

Reply:
I have also just realized that we are not running that test code really :-)
All good